extensions/dim-rename/src/main/java/io/druid/query/extraction/ClusterDimValRenameModule.java

```diff
@@ -0,0 +1,153 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Provides;
+import com.google.inject.name.Named;
+import io.druid.guice.LifecycleModule;
+import io.druid.guice.ManageLifecycle;
+import io.druid.guice.annotations.Json;
+import io.druid.initialization.DruidModule;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ *
+ */
+public class ClusterDimValRenameModule implements DruidModule
+{
+  private static final String preloadKey = "druid.query.rename.preload";
+  private static final String propertiesKey = "druid.query.rename.kafka.properties";
+
+  @Override
+  public List<? extends Module> getJacksonModules()
+  {
+    return ImmutableList.<Module>of(
+        new Module()
+        {
+          @Override
+          public String getModuleName()
+          {
+            return "druid-dim-rename";
+          }
+
+          @Override
+          public Version version()
+          {
+            return Version.unknownVersion();
+          }
+
+          @Override
+          public void setupModule(SetupContext setupContext)
+          {
+            setupContext.registerSubtypes(ClusterDimRenameFn.class);
+          }
+        }
+    );
+  }
+
+  @Provides
+  @Named("renameKafkaProperties")
+  public Properties getProperties(
+      @Json ObjectMapper mapper
+  )
+  {
+    String val = System.getProperty(propertiesKey);
+    if (val == null) {
+      return new Properties();
+    }
+    try {
+      Properties properties = new Properties();
+      properties.putAll(
+          mapper.<Map<String, String>>readValue(
+              val, new TypeReference<Map<String, String>>()
+          {
+          }
+          )
+      );
+      return properties;
+    }
+    catch (IOException e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  // JsonConfigProvider sucks at parameterized types
+  @Provides
+  @Named("preloadedTopics")
+  public List<String> getPreloadedTopics(
+      @Json ObjectMapper mapper
+  )
+  {
+    String val = System.getProperty(preloadKey);
```

Comment:
Get the Properties injected, not System.getProperty().  The bootstrap doesn't  guarantee that the property will be set as a System property, but it does guarantee that wherever it got the property from, it will be in the Properties object.